File: Cpp/53_Maximum_Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int result = INT_MIN;
        int sum = 0;
        for(auto num : nums) {
            sum += num;
            result = max(result, sum);
            sum = max(sum, 0);
        }
        
        return result;
    }
};

File: Kotlin/53_Maximum_Subarray.kt
class Solution {
    fun maxSubArray(nums: IntArray): Int {
        var result = nums.first()
        var sum = 0
        nums.forEach{ num ->
            sum += num
            result = maxOf(result, sum)
            sum = maxOf(sum, 0)
        }
        
        return result
    }
}